src/erasure-code/clay/ErasureCodePluginClay.cc

```diff
@@ -40,5 +40,10 @@ const char *__erasure_code_version() { return CEPH_GIT_NICE_VER; }
 int __erasure_code_init(char *plugin_name, char *directory)
 {
   auto& instance = ceph::ErasureCodePluginRegistry::instance();
-  return instance.add(plugin_name, new ErasureCodePluginClay());
+  auto plugin = new ErasureCodePluginClay();
```

Comment:
another way is to use a unique_ptr, like
```c++
auto plugin = std::make_unique<ErasureCodePluginClay>();
int r = instance.add(plugin_name, plugin);
if (r == 0) {
  plugin.release();
}
return r;
```

the goal is to avoid using raw pointer, and hence avoid using raw `new` and raw `delete`.

Reply:
Fixed.